bst_client_server/modules/zmq_mdlw/src/bst_kpsr_zmq_proxy_server.cpp

```diff
@@ -126,12 +124,16 @@ class ZmqLvcProxy : public ZmqProxy {
             return;              //  Interrupted
 
         if (items[0].revents & ZMQ_POLLIN) {
-            std::string topic = s_recv (_frontend);
-            std::string contents = s_recv (_frontend);
-            spdlog::info("{}new message received from server. Topic {}. Contents: {}", __PRETTY_FUNCTION__, topic, contents);
-            _cache[topic] = contents;
-            s_sendmore (_backend, topic);
-            s_send (_backend, contents);
+            zmq::message_t topicMsg;
+            zmq::message_t content;
+            _frontend.recv(topicMsg);
+            _frontend.recv(content);
+            std::string topic(static_cast<char*>(topicMsg.data()), topicMsg.size());
+            std::string contentString(static_cast<char*>(content.data()), content.size());
+            spdlog::info("{}new message received from server. Topic {}. Contents: {}", __PRETTY_FUNCTION__, topic, contentString);
+            _cache[topic] = contentString;
+            _backend.send(zmq::const_buffer(topic.data(), topic.size()), zmq::send_flags::sndmore);
```

Comment:
Instead of creating a new const_buffer, you can just reuse topicMsg directly.

Reply:
true, didnt notice. Thanks